tests/modules/programs/qcal/http-calendar.nix

```diff
@@ -0,0 +1,19 @@
+{ pkgs, lib, ... }:
+
+{
+  programs.qcal.enable = true;
+  accounts.calendar.accounts.test = {
+    qcal.enable = true;
+    remote = { url = "https://example.com/events.ical"; };
+  };
+
+  test.stubs = { qcal = { }; };
+
+  nmt.script = let
+    shared = (import ./shared.nix);
+    compare = shared.semanticallyCompare (lib.getExe pkgs.jq);
+  in ''
+    # Semantically compare the expected json output with the generated json
+    ${compare ./http-calendar.json-expected}
```

Comment:
I wouldn't worry about these semantic comparisons, the JSON output should be deterministic so you should be able to just put the expected output as-is in the result files.

Reply:
I see, that's nice. I feel like test cases are much more readable (and easier to write as well) if avoiding the minimized one-line json file though. I can of course take it out, but please also consider that side.

Reply:
In that case it may be better to make the module itself generate the pretty JSON file. E.g., by using `pkgs.formats.json { }`. Basically it would be to replace the line

        xdg.configFile."qcal/config.json".text = builtins.toJSON {

by something like

```
xdg.configFile."qcal/config.json".source = let
  jsonFormat = pkgs.formats.json { };
in jsonFormat.generate "qcal.json" {
```

Reply:
Implemented it this way. Thanks for the suggestion!